### DISCUSSION TITLE
cloudflare block

### DISCUSSION THREAD

**Discussion by pedroanhaia** [cloudbet/docs#77] [Q&A]
I am updating events every 15 minutes and new league events every hour. I was blocked by the cloudflare security system, how do I unblock and avoid further blocks?

**Comment by kgravenreuth** [comment]
Hi @pedroanhaia ,

There can be different reasons why you might get blocked by our security rules. Please review the list below:

- Access from restricted countries as per [Cloudbet Terms & Conditions](https://www.cloudbet.com/en/help/terms?section=71-your-account) is blocked.
- Requests without valid user agent
- Hitting our rate limits as per our [API Docs](https://www.cloudbet.com/api/?urls.primaryName=Trading%20B2B)

Please review your requests and try again. If the requests fail, please let us know with a screenshot of the error, thanks.

**Reply by bc-maker** [reply]
What are the valid user agents? 

I'm trying to place bets using Postman but have been blocked by Cloudflare also - How can I unblock?

**Reply by kgravenreuth** [reply]
Hi @bc-maker ,

Could you please post a screenshot of the error for us to investigate further, as it is difficult to pinpoint the error reasons without this information? Thanks.

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by bc-maker** [reply]
![image](https://github.com/Cloudbet/docs/assets/160312051/89c935c4-ca04-4c83-ad13-c4293e9b31c5)

**Reply by kgravenreuth** [reply]
Hi @bc-maker ,

Thanks for the screenshot. After review, we would like to point you to our list of blocked countries as mentioned in my message above.

> Access from restricted countries as per [Cloudbet Terms & Conditions](https://www.cloudbet.com/en/help/terms?section=71-your-account) is blocked.

Note that such country restrictions are imposed both at account registration as well as bet placement times and depends on the location of your API client. This is also mentioned in our [Trading API Docs](https://www.cloudbet.com/api/?urls.primaryName=Trading) 

Let me know if this clarifies your question, cheers.

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by bc-maker** [reply]
Hey yes I reviewed the list of countries before and can confirm I'm not in
the list. I thought I read somewhere this would give a different error if
it was hit anyway.

FYI I'm from New Zealand which isn't from the list

**Reply by kgravenreuth** [reply]
Hi @bc-maker ,

As discussed in more detail in #43 , auto-assigned IP addresses from cloud providers such as AWS could be detected as being from a restricted geography such as the USA. 

We would recommend you to follow your cloud provider's instructions on provisioning a static IP address explicitly in the geography where your service is meant to be provisioned. For example, please refer to the [AWS Documentation for Elastic IP Addresses](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html).

Let me know if this addresses your issue or if you need any additional help here.

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Comment by bc-maker** [comment]
Hey yes I reviewed the list of countries before and can confirm I'm not in
the list. I thought I read somewhere this would give a different error if
it was hit anyway

**Reply by kgravenreuth** [reply]
Hi @bc-maker ,

Please refer to my reply in the thread above about how to use cloud static IP addresses correctly, cheers.

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet
